lib/resolv.rb

```diff
@@ -2464,7 +2464,7 @@ class IPv6
     Regex_8HexLinkLocal = /\A
       [Ff][Ee]80
       (?::[0-9A-Fa-f]{1,4}){7}
-      %[0-9A-Za-z]+
+      %[-0-9A-Za-z._~]+
```

Comment:
```suggestion
      %[^\s\/:]+
```

Reply:
Thanks for pointing that out!  Truly crazy that it allows almost any character.  I think since `%` is the delimiter character according to RFC4007, we should not allow that.  What is the justification for not allowing whitespace, `/`, and `:`?  RFC4007 section 11.2 seems to allow implementations that support any non-null string that doesn't include the delimiter.  So `[^%]+` seems to be the widest interpretation.  What are your thoughts?

Reply:
I believe that's okay:

```
$ sudo ip l add 'test%1' type veth peer name dummy2
RTNETLINK answers: Invalid argument
```